android-core/src/main/java/com/mparticle/internal/ConfigManager.java

```diff
@@ -922,9 +924,11 @@ public boolean shouldTrigger(BaseMPMessage message) {
             isBackgroundAst = (message.getMessageType().equals(Constants.MessageType.APP_STATE_TRANSITION) && message.get(Constants.MessageKey.STATE_TRANSITION_TYPE).equals(Constants.StateTransitionType.STATE_TRANS_BG));
         } catch (JSONException ex) {
         }
+        if (isBackgroundAst) {
+            return false;
+        }
         boolean shouldTrigger = message.getMessageType().equals(Constants.MessageType.PUSH_RECEIVED)
-                || message.getMessageType().equals(Constants.MessageType.COMMERCE_EVENT)
-                || isBackgroundAst;
+                || message.getMessageType().equals(Constants.MessageType.COMMERCE_EVENT);//|| isBackgroundAst;
```

Comment:
`//|| isBackgroundAst;`

We should get rid of this extra commented out code

Reply:
Done